st2common/tests/unit/test_param_utils.py

```diff
@@ -362,15 +362,17 @@ def test_get_finalized_params_with_list(self):
             'a2': 'Test',
             'a3': 'Test2',
             'a4': '{{a1}}',
-            'a5': ['{{a2}}', '{{a3}}']
+            'a5': ['{{a2}}', '{{a3}}'],
+            'a6': [['{{r1}}', '{{a2}}'], ['{{a3}}', '{{a1}}']]
```

Comment:
I honestly think there is no way to sanely solve this in Jinja for non-homogenous types of items in the array. https://spacetelescope.github.io/understanding-json-schema/reference/array.html#list-validation
Though JSON schema allows heterogenous types inside the array, Jinja is the main problem here. The way we do Jinja rendering is we convert types to string and we convert strings back to appropriate types back based on user letting us know what type the variable is. This information is lost for list of heterogenous types where user doesn't specify what each item is like. We simply cannot solve this. @Kami Wdyt?